### PR TITLE
Graph3d tooltip styling

### PR DESCRIPTION
**Pull request by p-a**
PR for issue #2769 

Graph3D's option can now take a 'tooltipStyle' object with styling properties for 'content', 'line' and 'dot'.

### CODE REVIEW

**Comment by Tooa** [comment]
Thanks for your pull request! The network module uses a class name `network-tooltip` in the tooltip code. See [here](https://github.com/almende/vis/issues/626). Is there any benefit providing the styling as option parameter instead of using a css based approach like in the network module?

**Comment by p-a** [comment]
Well, my first thought was to just replace all hard coded styling altogether with a class, but then I realised that none of the Graph3D examples loads the bundled css file and doing such as thing would break things. Therefore I aimed for a more backwards compatible solution.

But yes, for a new major version a better solution would be to introduce a `graph3d-tooltip` class.
I can of course submit a new PR for such a change.

Any chance that the current PR could be applied for a new minor release since it doesn't break things?

**Comment by Tooa** [comment]
I see - good point. Then this approach is absolute fine for a minor/major version update. Let's keep it as it is and we include it in the next minor release. 

> I can of course submit a new PR for such a change.

I would love to see another pull request, which adds the same feature with the class thing. We are then able to merge this into our `5.0` branch.

File: lib/graph3d/Settings.js [hunk 1]
```diff
@@ -250,6 +250,15 @@ function setOptions(options, dst) {
   setSpecialSettings(options, dst);
 }
 
+/**
+ * Merge fields from src to dst
+ */
+function mergeFields(src, dst, fields) {
```
Comment (Tooa) [inline lib/graph3d/Settings.js]: I think something like this already exists in the utils [here](https://github.com/p-a/vis/blob/d4a8e878a36f26ea539a2bd15060eecb542a5bfa/lib/util.js#L198).

File: examples/graph3d/13_tooltip_styling.html [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      // create our graph
+      var container = document.getElementById('mygraph');
+      graph = new vis.Graph3d(container, data, options);
+
+      if (camera) graph.setCameraPosition(camera); // restore camera position
+
+      document.getElementById('style').onchange = drawVisualization;
+    }
+  </script>
+  <script src="../googleAnalytics.js"></script>
```
Comment (Tooa) [inline examples/graph3d/13_tooltip_styling.html]: As far as I know, we don't want `googleAnalytics` anymore ([reference](https://github.com/almende/vis/pull/2670)). Can you remove it?

File: examples/graph3d/13_tooltip_styling.html [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        width:  '600px',
+        height: '600px',
+        style: style,
+        showPerspective: true,
+        showLegend: true,
+        showGrid: true,
+        showShadow: false,
+
+        // Option tooltip can be true, false, or a function returning a string with HTML contents
+        //tooltip: true,
```
Comment (Tooa) [inline examples/graph3d/13_tooltip_styling.html]: Remove `tooltip: true,`

File: examples/graph3d/13_tooltip_styling.html [hunk 4]
```diff
@@ -0,0 +1,132 @@
+<!doctype html>
```
Comment (Tooa) [inline examples/graph3d/13_tooltip_styling.html]: This example is basically just a copy of [this](https://github.com/almende/vis/blob/develop/examples/graph3d/11_tooltips.html) one. Maybe it is better to adapt the other example for such a small change. What do you think?

File: docs/graph3d/index.html [hunk 5]
```diff
@@ -499,6 +499,35 @@ <h2 id="Configuration_Options">Configuration Options</h2>
     </tr>
 
     <tr>
+      <td>tooltipStyle</td>
+      <td>Object</td>
+      <td>{<br>
+        content&nbsp;:<br>{
```
Comment (Tooa) [inline docs/graph3d/index.html]: Do we really need No-break spaces here?

Comment (mojoaxel) [inline docs/graph3d/index.html]: @p-a Better use a <pre> block here.

File: examples/graph3d/13_tooltip_styling.html (hunk 4 above)
Comment (mojoaxel) [inline examples/graph3d/13_tooltip_styling.html]: I think so too. Feel free to adopt the existing example to your needs.

File: examples/graph3d/13_tooltip_styling.html (hunk 2 above)
Comment (mojoaxel) [inline examples/graph3d/13_tooltip_styling.html]: Yes please remove the analytics.

File: lib/graph3d/Graph3d.js [hunk 6]
```diff
@@ -67,6 +67,29 @@ var DEFAULTS = {
 
   style            : Graph3d.STYLE.DOT,
   tooltip          : false,
+
+  tooltipStyle     : {
```
Comment (mojoaxel) [inline lib/graph3d/Graph3d.js]: I haven't tested it yet. Please make sure the existing tooltips are basically not changed, or we have to consider this as a breaking change.

File: examples/graph3d/13_tooltip_styling.html (hunk 4 above)
Comment (p-a) [inline examples/graph3d/13_tooltip_styling.html]: ok. removed "my" example and altered the original a bit

File: lib/graph3d/Settings.js (hunk 1 above)
Comment (p-a) [inline lib/graph3d/Settings.js]: Done!. Used the util's methods instead

File: examples/graph3d/13_tooltip_styling.html (hunk 3 above)
Comment (p-a) [inline examples/graph3d/13_tooltip_styling.html]: done!

File: examples/graph3d/13_tooltip_styling.html (hunk 2 above)
Comment (p-a) [inline examples/graph3d/13_tooltip_styling.html]: Sorry, this was probably due to me branching of master initially...  But this file is now remoed altogether.

File: docs/graph3d/index.html (hunk 5 above)
Comment (p-a) [inline docs/graph3d/index.html]: Done!

File: lib/graph3d/Graph3d.js (hunk 6 above)
Comment (p-a) [inline lib/graph3d/Graph3d.js]: Tested all combos, if no `tooltipStyle` is provided the tooltips are not changed and are exactly as before.

Comment (p-a) [inline lib/graph3d/Graph3d.js]: @mojoaxel changes ok?

**Pull request merged**